best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,317 @@
+
+The DNS-SD advertisement MAY include a TXT record with key 'api_selector' and a string. The 'api_selector' key defines an arbitrary label that if present MUST be appended to the well-know EST path, for example `https://www.example.com/.well-known/est/<arbitrary label>/`. The arbitrary label is specified in the EST [RFC 7030][RFC-7030] and allows multiple EST Server instances on a single host to be used. If the 'api_selector' key is not present in the TXT record, the EST Client must not append anything to the well-know EST path.
+
+## EST Server Behaviour
+
+The EST server must be implemented in accordance with [RFC 7030][RFC-7030].
+
+### EST Server API
+
+The EST Server SHALL present an instance of the EST API as defined in [RFC 7030][RFC-7030].
+
+This MUST include the following API endpoints:
+
+| Operation                       | Operation path  |
+| ------------------------------- | --------------- |
+| Distribution of CA Certificates | /cacerts        |
+| Enrollment of Clients           | /simpleenroll   |
+| Re-enrollment of Clients        | /simplereenroll |
+
+
+### EST Server Authentication
+
+The EST Server MUST present a valid TLS Server Certificate signed by the CA for the realm to which it is issuing certificates.
+
+### EST Server - Client Authentication
+
+The EST Server MUST authenticate the EST Client that is requesting a TLS Certificate manually or automatically using a TLS Client Certificate.
+
+The EST Server MUST support using a TLS Client Certificate, presented during the TLS handshake by the EST Client to authenticate if the EST Client is trusted. The TLS Client Certificate can either be signed by the current CA or an externally trusted Root CA. The EST Server MUST check the validity of the EST CLients TLS Certificate before responding to its request. The EST Server MUST provide a method to load multiple trusted Root CA's, that are used to verify TLS Client Certificate.
+
+The EST Server MAY also support manual authentication of the EST Client if:
+- No TLS Client Certificate is presented during the TLS handshake
+- The TLS Client Certificate is not trusted
+- As an extra authentication step.
+The exact process for manual authentication will be implementation specific, but the EST Server MUST provide enough information to the operator so they can authenticate the EST Client. During the manual authentication the EST Server MUST respond with either HTTP 202 or HTTP 503 and the response must include a `Retry-After` header.
+
+The EST Server MUST be capable of issuing TLS Certificates signed with both RSA and ECDSA keys.
+
+The EST Server MUST return a TLS Certificate with the Extended Key Usage set for both TLS Server Authentication and TLS Client AUthentication as per [RFC 5280][RFC-5280]. This is to allow the TLS Certificate to be used both as a server certificate to authenticate NMOS APIs to NMOS Clients and for authentication of the EST Client to the EST Server during certificate renewal.
+
+## EST Client
```

Comment:
Do we need to say that the EST Client MUST support handling of the Retry-After procedure in case this is being used?

Reply:
Added description of handling HTTP 202 response, but not 503